### PR TITLE
fix crashes with invalid values for `svgReplaces` and `faces`

### PR DESCRIPTION
**Pull request by ArnoldSmith86**
It is not easy to make `svgReplaces` evaluate to `null` but it is possible with `inheritFrom`. This causes a crash which this PR fixes.

Similar behavior happens for `faces` where the code assumed it is an array. The resulting crash should now also be fixed.

Additionally this PR fixes a crash when indexing a variable that evaluates to `null` in a dynamic expression.

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: https://test.virtualtabletop.io/PR-1165/pr-test (or any other room on that server)

**Comment by rogerl50** [comment]
This appears to fix the problem that resulted in this PR.

**Review by rogerl50 — APPROVED** [review]
Approved through test of the original test case (only).

**Review by robartsd — CHANGES_REQUESTED** [review]

File: client/js/widgets/basicwidget.js [hunk 1]
```diff
@@ -72,29 +72,34 @@ class BasicWidget extends Widget {
...
+        await this.set('activeFace', Math.floor(this.get('activeFace') + (fC == 'random' ? Math.random()*99999 : 1)) % this.faces().length);
     }
   }
 
   getDefaultValue(property) {
-    if(property == 'faces' || property == 'activeFace' || !this.get('faces') || !this.get('faces')[this.get('activeFace')])
+    if(property == 'faces' || property == 'activeFace' || !this.faces() || !this.faces()[this.get('activeFace')])
       return super.getDefaultValue(property);
-    const d = this.get('faces')[this.get('activeFace')][property];
+    const d = this.faces()[this.get('activeFace')][property];
```
[the remark this answers is not in the document]
Comment (ArnoldSmith86) [reply client/js/widgets/basicwidget.js:95]: But the new `this.faces` should always return an array. That's what it's for. And the `if` two lines prior checks if `activeFace` is a valid index of `faces`.

**Review by robartsd — APPROVED** [review]

File: client/js/widgets/basicwidget.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ArnoldSmith86) [reply client/js/widgets/basicwidget.js:95]: Yeah, good catch. I just did a search and replace for `this.get('faces')`.

**Pull request merged**